accepted/2021/workloads/workload-installation.md

```diff
@@ -0,0 +1,351 @@
+
+The IAL is an abstraction that is internal to the .NET SDK.  It is not intended as a public extensibility mechanism to allow supporting arbitrary installers, and we may make breaking changes to its API as needed.
+
+# Supported installation types
+
+We plan to (eventually) support the following installation types.  Some of them may not be supported in .NET 6 if we don't have workloads in .NET 6 that are supported on the corresponding platforms.
+
+## File based installation
+
+In this implementation of the IAL, the .NET SDK itself manages workload installations directly by writing files to the dotnet folder.  No package management or external installation technology is used. This is generally for copies of the .NET SDK which were not installed via an OS installer, for example those that are installed via the dotnet-install scripts, or via unzipping archives of the .NET SDK.
+
+Other terms we have at times used for this installation type: .NET SDK managed, xcopy, universal installer, files on disk, NuGet installer
+
+## Visual Studio based installation
+
+This is an installation of the .NET SDK installed and managed by the Visual Studio installer.  The .NET SDK workloads will have corresponding Visual Studio workloads or components, which will install workload packs via Windows MSIs.  Note that the Visual Studio workloads or components should also include the necessary Visual Studio components (designers, project system, etc) to support the workload, in addition to the .NET SDK workload packs.
+
+## MSI based installation
+
+This is an installation of the .NET SDK installed on Windows via the .NET SDK installer bundle, which is the .exe which can be downloaded on the .NET SDK download pages.  The bundle installs the same MSIs for the .NET SDK as Visual Studio does, and the workload installer will likewise install the same MSIs for workload packs as the Visual Studio installer.
+
+The same dotnet folder may have different versions of the SDK installed by both the Visual Studio installer and the Standalone installer, which may share the same versions of some components.  For the .NET SDK, the underlying MSIs are ref-counted so that they are uninstalled when there are no more installations that depend on them.  We will need to make sure that workload installation and management also works well when there are different install types coexisting.
+
+## Package manager installation
+
+We have not yet settled on how workload installation will work when the .NET SDK is installed via a Linux package manager such as APT or RPM.
+
+## For .NET 6
+
+The only workload that will be supported on Linux in .NET 6 will be Blazor AoT.  For .NET 6, we will recommend that developers using a .NET SDK installed via a package manager run `dotnet workload install` under `sudo`.  This will use the file based installation mechanism, ie it will download the workload packs and copy them to the .NET SDK folder.
```

Comment:
The install location used by package managers is **not** a location a user should add additional packages to.